File: Leetcode/C++/My Calendar I.py
class MyCalendar:

    def __init__(self):
        self.l = 0
        self.res = []

    def book(self, start, end):
        """
        :type start: int
        :type end: int
        :rtype: bool
        """
        for left, right in self.res:
        	if not (end <= left or start >= right): return False
        self.res.append((start, end))
        return True



# Your MyCalendar object will be instantiated and called as such:
# obj = MyCalendar()
# param_1 = obj.book(start,end)

File: Leetcode/C++/My Calendar I.cpp
class MyCalendar {
public:
    MyCalendar() {
        
    }
    vector<vector<int>>res;
    bool book(int start, int end) {
        for(int i = 0; i< res.size(); i++){
        	int s = res[i][0];
        	int e = res[i][1];
        	if (!(start >= e || end <= s)) return false;
        }
        res.push_back({start, end});
        return true;
    }
};

/**
 * Your MyCalendar object will be instantiated and called as such:
 * MyCalendar obj = new MyCalendar();
 * bool param_1 = obj.book(start,end);
 */